lib/nolibc/stdio.c

```diff
@@ -541,3 +543,78 @@ int puts(const char *s)
 {
 	return fputs_internal(s, stdout, 1);
 }
+
+#if CONFIG_LIBVFSCORE
+
+int __fmodeflags(const char *mode)
+{
+	int flags;
+
+	if (strchr(mode, '+'))
+		flags = O_RDWR;
+	else if (*mode == 'r')
+		flags = O_RDONLY;
+	else
+		flags = O_WRONLY;
+	if (strchr(mode, 'x'))
+		flags |= O_EXCL;
+	if (strchr(mode, 'e'))
+		flags |= O_CLOEXEC;
+	if (*mode != 'r')
+		flags |= O_CREAT;
+	if (*mode == 'w')
+		flags |= O_TRUNC;
+	if (*mode == 'a')
+		flags |= O_APPEND;
+	return flags;
+}
+
+struct vfscore_file *fopen(const char *pathname, const char *mode)
+{
+	struct vfscore_file *f;
+	int flags;
+	int fd;
+
+	if (!strchr("rwa", *mode)) {
+		errno = EINVAL;
+		return 0;
+	}
+
+	flags = __fmodeflags(mode);
+	fd = open(pathname, flags, 0666);
```

Comment:
Could you use something from mode instead of hardcoding `0666` here :thinking:

Reply:
`checkpatch.uk` recommended this way of doing it so I went along with it.